src/atomate2/common/flows/mpmorph.py

```diff
@@ -0,0 +1,493 @@
+        MDMaker to generate the production run(s)
+    quench_maker :  SlowQuenchMaker or FastQuenchMaker or None
+        SlowQuenchMaker - MDMaker that quenches structure from
+            high to low temperature
+        FastQuenchMaker - DoubleRelaxMaker + Static that
+            "quenches" structure at 0K
+    """
+
+    name: str = "Base MPMorph MD"
+    convergence_md_maker: Maker | None = None  # check logic on this line
+    # May need to fix next two into ForceFieldMDMakers later..)
+    production_md_maker: Maker | None = None
+    quench_maker: FastQuenchMaker | SlowQuenchMaker | None = None
+
+    def make(
+        self,
+        structure: Structure,
+        prev_dir: str | Path | None = None,
+    ) -> Flow:
+        """
+        Create a flow with MPMorph molecular dynamics (and relax+static).
+
+        By default, production run is broken up into multiple smaller steps.
+        Converegence and quench are optional and may be used to equilibrate
+        the wt cell volume (useful for high temperature production runs of
+        structures extracted from Materials Project) and to quench the
+        structure from high to low temperature (e.g. amorphous structures).
+
+        Parameters
+        ----------
+        structure : .Structure
+            A pymatgen structure object.
+        prev_dir : str or Path or None
+            A previous VASP calculation directory to copy output files from.
+
+        Returns
+        -------
+        Flow
+            A flow containing series of molecular dynamics run (and relax+static).
+        """
+        flow_jobs = []
+
+        if self.convergence_md_maker is not None:
+            convergence_flow = self.convergence_md_maker.make(
+                structure, prev_dir=prev_dir
+            )
+            flow_jobs.append(convergence_flow)
+
+            # convergence_flow only outputs a structure
+            structure = convergence_flow.output
+
+        self.production_md_maker.name = self.name + " production run"
+        production_run = self.production_md_maker.make(
+            structure=structure, prev_dir=prev_dir
+        )
+        flow_jobs.append(production_run)
+
+        if self.quench_maker:
+            quench_flow = self.quench_maker.make(
+                structure=production_run.output.structure,
+                prev_dir=production_run.output.dir_name,
+            )
+            flow_jobs += [quench_flow]
+
+        return Flow(
+            flow_jobs,
+            output=production_run.output,
+            name=self.name,
+        )
+
+    @classmethod
+    def from_temperature_and_steps(
+        cls,
+        temperature: float,
+        n_steps_convergence: int = 5000,
+        n_steps_production: int = 10000,
+        end_temp: float | None = None,
+        md_maker: Maker = None,
+        quench_maker: FastQuenchMaker | SlowQuenchMaker | None = None,
+    ) -> Self:
+        """
+        Create an MPMorph flow from a temperature and number of steps.
```

Comment:
looks like this is missing an `@abstractmethod` decorator?

Reply:
Added this decorator to a few other places where I can see it's needed (eos common flows, ASE base classes, OpenMM base class)